### PR TITLE
Save memory by interning strings and using trove

### PR DESCRIPTION
**Pull request by donnerpeter**

### CODE REVIEW

File: languagetool-core/src/main/java/org/languagetool/tagging/ManualTagger.java [hunk 1]
```diff
@@ -79,7 +79,8 @@ public ManualTagger(InputStream inputStream, boolean internTags) throws IOExcept
         lemma = interned.computeIfAbsent(lemma, Function.identity());
 
         String tag = parts[2].trim();
-        map.computeIfAbsent(form, __ -> new ArrayList<>()).add(new TaggedWord(lemma, internTags ? tag.intern() : tag));
+        String internedTag = internTags ? tag.intern() : interned.computeIfAbsent(tag, Function.identity());
```
Comment (donnerpeter) [inline languagetool-core/src/main/java/org/languagetool/tagging/ManualTagger.java:82]: You might decide that you don't need `internTags` parameter anymore

**Comment by donnerpeter** [comment]
~5MB on de+en

**Comment by danielnaber** [comment]
@donnerpeter This might cause a rare issue documented at https://github.com/languagetool-org/languagetool/issues/5224, do you have any idea?

**Comment by donnerpeter** [comment]
This can be caused by a JMM violation. I'll take a look.

**Pull request merged**